### DISCUSSION TITLE
Catching Event coming from a Geoman Layer Or General Map

### DISCUSSION THREAD

**Discussion by cubecleveland** [geoman-io/leaflet-geoman#1104] [Questions & Help]
Hi, is there any way to know if a clikc event came form a geoman layer without attaching an event handler for every layer.
Basically since the ui  also enable creating of new layers i want to see if a clikc event any where on the map is coming from a drawn layer but withou putting the handler specifically on the layer , since clicking outside actually toggle a different state in the UI.

thanks!@

**Comment by Falke-Design** [comment]
If you have the layer object, you can check for the `pm` property if it is a geoman layer `layer.pm`.

**Comment by cubecleveland** [comment]
Can that be tapped from the main map instance click event ?

what im trying to acheive is to capture all clicks and see if theyc ame from Geoman layer or not....
If they dont i do something else... than if thy did

**Reply by Falke-Design** [reply]
No there is no built in function in Leaflet and Geoman for this. You need to make this by your self, get the coordinates of the click and then check which layer is at this coords.
Another way would be to add click listeners to the layers and then fireing from there a map click event with a specific property in the payload. Then you know that it comes from a layer.

Maybe a other thing is your solution. When clicking to an UI Element which is not on the map, it should not fire a click event. For this you can use [L.DomEvent.disableClickPropagation(DOM-Element)](https://leafletjs.com/reference.html#domevent-disableclickpropagation), then the click event is not fired on the map.
